app/src/main/java/com/example/mytestproject/data/repository/WeatherRepository.kt

```diff
@@ -0,0 +1,12 @@
+package com.example.mytestproject.data.repository
+
+import com.example.mytestproject.data.network.response.WeatherDataApi
+import io.reactivex.rxjava3.core.Single
+
+interface WeatherRepository {
+    fun getWeatherData(
+        city: String,
+        days: Int,
+        degreeType: String
+    ): Single<WeatherDataApi>
```

Comment:
WeatherDataApi shouldn't be used outside of repository. This data class is needed only to get data from the API. 
Let's create new data class WeatherData that will be used in UI part. Structure of this data can be the same as WeatherDataApi. It's normal.

Repository should do a mapping from WeatherDataApi to WeatherData. To do a mapping we need to create and use a Mapper. 

As a result, repository should give us Single<WeatherData> instead of Single<WeatherDataApi>.